templates/go-client/schema.tmpl

```diff
@@ -11,7 +11,17 @@ type {{ .GetName | SanitiseName }} {{ . | SanitiseType }}
         {{ if eq ($schema | SanitiseType) "struct" -}}
             {{ $name | ToTitle }} {{ $schema.GetName | SanitiseName }}
         {{- else -}}
-            {{ $name | ToTitle }} {{ $schema | SanitiseType }} `json:"{{ $name }}"`
+            {{- $required := false }}
+            {{- range $requiredProperty := $schema.Required }}
+                {{- if eq $name $requiredProperty }}
+                    {{- $required = true }}
+                {{- end }}
+            {{- end }}
+            {{- if eq $required false -}}
+                {{ $name | ToTitle }}  {{ if eq ($schema | SanitiseType) "bool" }}*{{ end }}{{ $schema | SanitiseType }} `json:"{{ $name }},omitempty"`
```

Comment:
Why not treat all optional types as pointers? Simplifies the template and is more spec compliant.

Reply:
I'd prefer to put that in an issue and expand to all other types. 
It's not entirely justified for all types IMO (for example, strings) and using pointers everywhere causes the consumer of these generated clients to have to check for nil everywhere. 
I'd rather sacrifice simplicity in the templates if it means that the consumer of can has a simpler experience using the generated code overall.

Reply:
With a bool I see `nil` as allowing for a third state (that state being representative of _nothing_) as a result of us being disallowed of a state that's representative of _nothing) without the use of `nil`. With a string, we have the option of representing a _nothing_ state through the use of an empty string or `""` which isn't granted to us by the bool type in go.

Reply:
I'm not beholden to this view, and could happily set up some code to make your suggestion a reality, but I'd prefer not to do it in this PR

Reply:
Having a rule within the development of can like _any optional value should be represented as a pointer for the indication of nothing through the use of `nil`_ is definitely something I could get behind, but it needs to be established first.

Reply:
Cool. Not to worry. I'll link this thread in the issue I create.
Resolving for now.

PS: empty strings are still valid api parameter values. So there still would be a need for nil string parameters when they are optional. Think about setting the description on your project to "" when you create an openstack project as a contrived example.

Still, something for the other issue.

Reply:
Rules like this would allow us to set principles that sensibly set up broad expectations in our users (through their inclusion in our documentation) while simultaneously providing broad guidelines for contributors to the codebase. 

The one immediate drawback that I see is that a principle set so broad could be used to motivate an unforgiving attitude to any edge case that may not fit the guideline